skins/laika/src/components/pages/donation_form/PaymentBankData.vue

```diff
@@ -52,12 +52,31 @@ export default Vue.extend( {
 		},
 	},
 	methods: {
-		setAccountId(): void {
-			this.$store.dispatch( action( NS_PAYMENT, setAccountId ), this.$data.accountId );
+		validate() {
+			if ( this.isAccountIdEmpty() ) {
+				return;
+			}
+			if ( this.looksLikeIban() ) {
+				this.$store.dispatch( action( NS_PAYMENT, setBankData ), this.$data );
+				return;
+			}
+			this.$store.dispatch( action( NS_PAYMENT, setLegacyBankdata ), this.$data );
```

Comment:
The `dispatch` method returns a promise. You can use that to update the bank data fields when the server returns a success response (especially relevant for the account number -> BIC conversion).